### PR TITLE
LargeantAnalysis module added

### PR DESCRIPTION
**Pull request by szhang17phys**
Modified version of previous AnalysisExample module

### CODE REVIEW

**Comment by szhang17phys** [comment]
Old AnalysisExample module is located at https://github.com/DUNE/lbnecode, which had not been updated for 9 years. The old AnalysisExample is based on previous LArG4. While this new LargeantAnalysis module is based on the LArG4 Main.

File: duneopdet/OpticalDetector/LargeantAnalysis_module.cc [hunk 1]
```diff
@@ -0,0 +1,801 @@
...
+// <https://cdcvs.fnal.gov/redmine/projects/larsoftsvn/wiki/Using_the_Framework>
+// for a description of the ART classes used here.
+
+// Almost everything you see in the code below may have to be changed
+// by you to suit your task. The example task is to make histograms
+// and n-tuples related to dE/dx of particle tracks in the detector.
+
+// As you try to understand why things are done a certain way in this
+// example ("What's all this stuff about 'auto const&'?"), it will help
+// to read ADDITIONAL_NOTES.txt in the same directory as this file.
```
Comment (lpaulucc) [inline duneopdet/OpticalDetector/LargeantAnalysis_module.cc:15]: Does this file exist in duneopdet?

File: duneopdet/OpticalDetector/LargeantAnalysis_module.cc [hunk 2]
```diff
@@ -0,0 +1,801 @@
...
+
+//Sep 24, 2024
```
Comment (lpaulucc) [inline duneopdet/OpticalDetector/LargeantAnalysis_module.cc:17]: Add your name and contact info here.

File: duneopdet/OpticalDetector/LargeantAnalysis_module.cc [hunk 3]
```diff
@@ -0,0 +1,801 @@
...
+      // middle of a job; e.g., if the user changes parameter values in an
+      // interactive event display.
+      void reconfigure(fhicl::ParameterSet const& pset);
+
+      // The analysis routine, called once per event. 
+      void analyze (const art::Event& evt); 
+
+    private:
+
+      // The stuff below is the part you'll most likely have to change to
```
Comment (lpaulucc) [inline duneopdet/OpticalDetector/LargeantAnalysis_module.cc:95]: You created a new module, this is not an example anymore

File: duneopdet/OpticalDetector/LargeantAnalysis_module.cc [hunk 4]
```diff
@@ -0,0 +1,801 @@
...
+      // go from this custom example to your own task.
+
+      // The parameters we'll read from the .fcl file.
```
Comment (lpaulucc) [inline duneopdet/OpticalDetector/LargeantAnalysis_module.cc:98]: You did not create a fhicl to go together with this module!

File: duneopdet/OpticalDetector/LargeantAnalysis_module.cc [hunk 5]
```diff
@@ -0,0 +1,801 @@
...
+    // we write a variable, we give the address of the variable to
+    // TTree::Branch.
+    fSimulationNtuple->Branch("Event",       &fEvent,          "Event/I");
+    fSimulationNtuple->Branch("SubRun",      &fSubRun,         "SubRun/I");
+    fSimulationNtuple->Branch("Run",         &fRun,            "Run/I");
+    fSimulationNtuple->Branch("TrackID",     &fTrackID,        "TrackID/I");
+    fSimulationNtuple->Branch("PDG",         &fPDG,            "PDG/I");
+    // When we write arrays, we give the address of the array to
+    // TTree::Branch; in C++ this is simply the array name.
+    fSimulationNtuple->Branch("StartXYZT",   fStartXYZT,       "StartXYZT[4]/D");
```
Comment (lpaulucc) [inline duneopdet/OpticalDetector/LargeantAnalysis_module.cc:213]: Are you sure those variables are not already in dune10kt_analysistree?

**Review by lpaulucc — COMMENTED** [review]
Hi Shu, I think the code could be cleaned a lot, there are large parts which are commented out. You are missing a fhicl file for your module but the main thing is: are you sure we cannot extract the info directly from the default analysis tree?

**Comment by tomjunk** [comment]
If the purpose is to provide an example, then there is a maintained version in larexamples

https://github.com/LArSoft/larexamples/tree/develop/larexamples/AnalysisExample

I think that's where the one in lbnecode was copied from.

What is added/modified here to make it useful for duneopdet, specifically?

**Pull request closed without merging**